crates/test-programs/wasi-tests/src/lib.rs

```diff
@@ -1,30 +1,34 @@
 pub mod utils;
 pub mod wasi_wrappers;
 
-use libc;
-use std::ffi::CString;
-use std::io;
 use wasi_old::wasi_unstable;
 
 /// Opens a fresh file descriptor for `path` where `path` should be a preopened
 /// directory. This is intended to be used with `wasi_unstable`, not with
 /// `wasi_snapshot_preview1`. This is getting phased out and will likely be
 /// deleted soon.
 pub fn open_scratch_directory(path: &str) -> Result<wasi_unstable::Fd, String> {
-    // Open the scratch directory.
-    let dir_fd: wasi_unstable::Fd = unsafe {
-        let cstr = CString::new(path.as_bytes()).unwrap();
-        libc::open(cstr.as_ptr(), libc::O_RDONLY | libc::O_DIRECTORY)
-    } as wasi_unstable::Fd;
+    unsafe {
+        for i in 3.. {
+            let stat = match wasi_unstable::fd_prestat_get(i) {
+                Ok(s) => s,
+                Err(_) => break,
+            };
+            if stat.pr_type != wasi::PREOPENTYPE_DIR {
+                continue;
+            }
+            let mut dst = Vec::with_capacity(stat.u.dir.pr_name_len);
+            if wasi::fd_prestat_dir_name(i, dst.as_mut_ptr(), dst.capacity()).is_err() {
```

Comment:
Hmm, is it OK to mix `wasi_unstable` and `wasi` for the syscalls here? I mean, first, we call `wasi_unstable::fd_prestat_get` just to later on call `wasi::fd_prestat_dir_name`. It might be OK, but definitely looks confusing!

Reply:
Oh oops this was a mistake! I'll switch this as well to match the above